standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/client/TestTablesGetExists.java

```diff
@@ -402,6 +393,134 @@ public void testGetTableObjectsByName() throws Exception {
+    request.setDbName(DEFAULT_DATABASE);
+
+    List<Table> tables = client.getTableObjectsByRequest(request);
+
+    Assert.assertEquals("Found tables", 2, tables.size());
+  }
+
+  @Test
+  public void testGetTableObjectsWithNonExistentColumn() throws Exception {
+    List<String> tableNames = new ArrayList<>();
+    tableNames.add(testTables[0].getTableName());
+    tableNames.add(testTables[1].getTableName());
+
+    GetTablesRequest request = new GetTablesRequest();
+    request.setProjectionSpec(new GetProjectionsSpec());
+    request.setTblNames(tableNames);
+    request.setDbName(DEFAULT_DATABASE);
+
+    GetProjectionsSpec projectSpec = request.getProjectionSpec();
+    List<String> projectedFields = Arrays.asList("Invalid1");
+    projectSpec.setFieldList(projectedFields);
+
+    Assert.assertThrows(Exception.class, ()->client.getTableObjectsByRequest(request));
+  }
+
+
+  @Test
+  public void testGetTableObjectsWithNonExistentColumns() throws Exception {
+    List<String> tableNames = new ArrayList<>();
+    tableNames.add(testTables[0].getTableName());
+    tableNames.add(testTables[1].getTableName());
+
+    GetTablesRequest request = new GetTablesRequest();
+    request.setProjectionSpec(new GetProjectionsSpec());
+    request.setTblNames(tableNames);
+    request.setDbName(DEFAULT_DATABASE);
+
+    GetProjectionsSpec projectSpec = request.getProjectionSpec();
+    List<String> projectedFields = Arrays.asList("Invalid1", "Invalid2");
+    projectSpec.setFieldList(projectedFields);
+
+    Assert.assertThrows(Exception.class, ()->client.getTableObjectsByRequest(request));
+  }
+
+  @Test
+  public void testGetTableObjectsWithEmptyProjection() throws Exception {
+    List<String> tableNames = new ArrayList<>();
+    tableNames.add(testTables[0].getTableName());
+    tableNames.add(testTables[1].getTableName());
+
+    GetTablesRequest request = new GetTablesRequest();
+    request.setProjectionSpec(new GetProjectionsSpec());
+    request.setTblNames(tableNames);
+    request.setDbName(DEFAULT_DATABASE);
+
+    GetProjectionsSpec projectSpec = request.getProjectionSpec();
+    List<String> projectedFields = Arrays.asList();
+    projectSpec.setFieldList(projectedFields);
+
+    List<Table> tables = client.getTableObjectsByRequest(request);
+
+    Assert.assertEquals("Found tables", 0, tables.size());
+  }
+
+  @Test
+  public void testGetTableObjectsWithProjectionOfMultipleField() throws Exception {
+    List<String> tableNames = new ArrayList<>();
+    tableNames.add(testTables[0].getTableName());
+    tableNames.add(testTables[1].getTableName());
+
+    GetTablesRequest request = new GetTablesRequest();
+    request.setProjectionSpec(new GetProjectionsSpec());
+    request.setTblNames(tableNames);
+    request.setDbName(DEFAULT_DATABASE);
+
+    GetProjectionsSpec projectSpec = request.getProjectionSpec();
+    List<String> projectedFields = Arrays.asList("database", "tableName", "createTime", "lastAccessTime");
```

Comment:
can we add some nested single valued fields here too. Something other than sd.location? like for example. sd.serdeInfo.serializationLib? Also, what is the expected behavior if the user adds a multi-valued field like sd.parameters. Can we have some tests around that too.